.kiro/specs/tls-rustls-migration/design.md

```
@@ -2,218 +2,464 @@
+        let ssl_opts = SslOpts::default().with_rustls_client_config(client_config);
+
+        Ok(Some(ssl_opts))
     }
 }
 ```
 
-### Migration Compatibility Layer
+### Custom Certificate Verifiers
+
+#### Hostname Skip Verifier
 
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)